File: src/ProjectR.Scripting/content/scripts/spells/RagingFirestorm.lua
function GetTargetType()
    return Enemies
end

function GetName()
return "Raging Firestorm"
end

function GetDescription()
return "Summon a raging firestorm that engulfs all enemies.\nReduces enemies AD & MD."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .6
end

function GetDelay()
  return .3
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { WND, FIR }
end

function SpellEffect()
  local damage = (3.375 * aMD * ((aFIR + aWND)/200) - 1.25 * dMR) * (200/(dFIR + dWND))  
  d:BuffStat(AD, -.12)
  d:BuffStat(MD, -.12)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/RagingFireStorm.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class RagingFireStorm : SpellScriptBase
    {
        public override string Name { get { return "Raging Firestorm"; } }
        public override string Description { get { return "Summon a raging firestorm that engulfs all enemies.\nReduces enemies AD & MD."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.WND, EleMastery.FIR, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 60; } }
        public override double Delay { get { return .3; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (3.375 * cMD * ((cFIR + cWND) / 200) - 1.25 * tMR) * (200 / (tFIR + tWND));
            BuffStat(Stat.AD, -.22);
            BuffStat(Stat.MD, -.22);
            DealDamage(damage);
        }
    }
}